### PR TITLE
Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL

### PR DESCRIPTION
**Pull request by osalpekar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#41076 Modularize parsing NCCL_BLOCKING_WAIT in ProcessGroupNCCL**

Modularizes Parsing the NCCL_BLOCKING_WAIT environment variable in the ProcessGroupNCCL Constructor.

Differential Revision: [D22401225](https://our.internmc.facebook.com/intern/diff/D22401225/)

### CODE REVIEW

File: torch/lib/c10d/ProcessGroupNCCL.cpp [hunk 1]
```diff
@@ -271,19 +286,8 @@ ProcessGroupNCCL::ProcessGroupNCCL(
       ncclCommCounter_(0),
       terminateWatchdog_(false),
       opTimeout_(opTimeout) {
```
Comment (osalpekar) [inline torch/lib/c10d/ProcessGroupNCCL.cpp:288]: We can alternately set `blockingWait_` from the `ProcessGroupNCCL` ctor initializer list and handle all exception cases in the helper - would that be better style-wise?

Comment (jiayisuse) [reply torch/lib/c10d/ProcessGroupNCCL.cpp:288]: Current solution looks good to me. Initialization list tends to do trivial initialization.

**Review by jiayisuse — APPROVED** [review]

**Comment by jiayisuse** [comment]
Seems CI fails to download cuda pkgs
```
Failed to fetch https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/Packages  Writing more data than expected (1196016 > 1190062)
```

**Comment by osalpekar** [comment]
> Seems CI fails to download cuda pkgs
> 
> ```
> Failed to fetch https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/Packages  Writing more data than expected (1196016 > 1190062)
> ```

I'll try rebasing, seems like some flakiness in the CI

**Pull request closed without merging**